### ISSUE TITLE
Exclude Main method from "Add null check" refactoring

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#46827]
```csharp
public static class Program
{
    public static void Main(string[] args) // CTRL + . on args suggests to "Add null check"
    {
    }
}
```

The analyzer can special-case the Main method in that case and don't emit a diagnostic.

**Comment by Youssef1313** [comment]
Sorry, IDE0060 isn't related to null, it's for unused argument. I meant a refactoring that suggests to validate `args` parameter.

**Comment by sharwell** [comment]
IMO the diagnostic is correct here. `Main` can be called from user code the same way any other method can be called (i.e. it is not required to be an application entry point). I've seen several command line applications in the past that allow but do not require `Main` to be the _original_ entry point in order to still be used.

**Comment by sharwell** [comment]
Tentatively marking as **By Design**; will update with the conclusion following design review.

**Comment by Youssef1313** [comment]
> IMO the diagnostic is correct here. Main can be called from user code the same way any other method can be called (i.e. it is not required to be an application entry point). I've seen several command line applications in the past that allow but do not require Main to be the original entry point in order to still be used.

Yes I agree, I missed the case where the Main can be called. However, what if it's `private`?

**Comment by sharwell** [comment]
> I never thought of referencing an executable to another project.

Referencing an executable works exactly the same way as referencing a class library. If you didn't stop to look at the file extensions, you might never even notice the difference. :)

**Comment by jasonmalinowski** [comment]
We discussed this in our design meeting and agreed to leave this as is, and count the bug as by design.
